packages/opentelemetry-plugin-postgres/src/utils.ts

```diff
@@ -0,0 +1,155 @@
+}
+
+// Private helper function to start a span
+function pgStartSpan(
+  tracer: Tracer,
+  client: pgTypes.Client & PgClientExtended,
+  name: string
+) {
+  const jdbcString = getJDBCString(client.connectionParameters);
+  return tracer.startSpan(name, {
+    kind: SpanKind.CLIENT,
+    parent: tracer.getCurrentSpan() || undefined,
+    attributes: {
+      [AttributeNames.COMPONENT]: PostgresPlugin.COMPONENT, // required
+      [AttributeNames.DB_INSTANCE]: client.connectionParameters.database, // required
+      [AttributeNames.DB_TYPE]: PostgresPlugin.DB_TYPE, // required
+      [AttributeNames.PEER_ADDRESS]: jdbcString, // required
+      [AttributeNames.PEER_HOSTNAME]: client.connectionParameters.host, // required
+      [AttributeNames.PEER_PORT]: client.connectionParameters.port,
+      [AttributeNames.DB_USER]: client.connectionParameters.user,
+    },
+  });
+}
+
+// Queries where args[0] is a QueryConfig
+export function handleConfigQuery(
+  this: pgTypes.Client & PgClientExtended,
+  tracer: Tracer,
+  ...args: unknown[]
+) {
+  const argsConfig = args[0] as PgPluginQueryConfig;
+
+  // Set child span name
+  const queryCommand = getCommandFromText(argsConfig.name || argsConfig.text);
+  const name = PostgresPlugin.BASE_SPAN_NAME + ':' + queryCommand;
+  const span = pgStartSpan(tracer, this, name);
+
+  // Set attributes
+  span.setAttribute(AttributeNames.DB_STATEMENT, argsConfig.text);
+  if (argsConfig.values instanceof Array) {
+    span.setAttribute(
+      AttributeNames.PG_VALUES,
+      arrayStringifyHelper(argsConfig.values)
+    );
+  }
+  // Set plan name attribute, if present
+  if (argsConfig.name) {
+    span.setAttribute(AttributeNames.PG_PLAN, argsConfig.name);
+  }
+
+  return span;
+}
+
+// Queries where args[1] is a 'values' array
+export function handleParameterizedQuery(
+  this: pgTypes.Client & PgClientExtended,
+  tracer: Tracer,
+  ...args: unknown[]
+) {
+  // Set child span name
+  const queryCommand = getCommandFromText(args[0] as string);
+  const name = PostgresPlugin.BASE_SPAN_NAME + ':' + queryCommand;
+  const span = pgStartSpan(tracer, this, name);
+
+  // Set attributes
+  span.setAttribute(AttributeNames.DB_STATEMENT, args[0]);
+  if (args[1] instanceof Array) {
+    span.setAttribute(AttributeNames.PG_VALUES, arrayStringifyHelper(args[1]));
+  }
+
+  return span;
+}
+
+// Queries where args[0] is a text query and 'values' was not specified
+export function handleTextQuery(
+  this: pgTypes.Client & PgClientExtended,
+  tracer: Tracer,
+  ...args: unknown[]
+) {
+  // Set child span name
+  const queryCommand = getCommandFromText(args[0] as string);
+  const name = PostgresPlugin.BASE_SPAN_NAME + ':' + queryCommand;
+  const span = pgStartSpan(tracer, this, name);
+
+  // Set attributes
+  span.setAttribute(AttributeNames.DB_STATEMENT, args[0]);
+
+  return span;
+}
+
+export function patchCallback(
+  span: Span,
+  cb: PostgresCallback
+): PostgresCallback {
+  const originalCb = cb;
```

Comment:
Why this additional variable? Can this line not be removed and on line 153 just `cb.call(...`?